### PR TITLE
#2066 doctest update losses

### LINKED ISSUE #2066: Testable docstrings
https://github.com/tensorflow/addons/pull/677 is merged, so we are happy to introduce doctest on the docstring. This feature is awesome that can avoid wrong example usage.

Because the change might be small, please submit one sub-module, for example, activations, per PR. For more details, please refer to https://www.tensorflow.org/community/contribute/docs_ref.

Note that available namespace in our repo is `tf` for `tensorflow`, `np` for `numpy` and `tfa` for `tensorflow_addons`.

- [x] activations: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/activations
- [ ] callbacks: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/callback
- [ ] image: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/image
- [x] layers: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/layers
- [x] losses: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/losses
- [x] metrics: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/metrics
- [ ] optimizers: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/optimizers
- [x] rnn: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/rnn
- [ ] seq2seq: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/seq2seq
- [ ] text: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/text

It's also really appreciated that one can help improve our doc based on https://www.tensorflow.org/community/contribute/docs_style when scrutinizing docstrings.

### PR DESCRIPTION
**Pull request by Harsh188**
# Description

Adding testable docstrings to `losses`.
Fixes #2066

Changes made to:
- `quantiles.py`
- `npairs.py`
- `kappa_loss.py`
- `giou_loss.py`
- `focal_loss.py`
- `contrastive.py`

## Type of change

- [x] Updated or additional documentation

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [x] I have made corresponding changes to the documentation

# How Has This Been Tested?
- Used pylint with TensorFlow's custom style definition

### CODE REVIEW

**Comment by Harsh188** [comment]
Looks like the checks are failing because I forgot to run `cd .git/hooks && ln -s -f ../../tools/pre-commit.sh pre-commit` before commiting my files. Please let me know how to fix this.

**Comment by seanpmorgan** [comment]
> Looks like the checks are failing because I forgot to run `cd .git/hooks && ln -s -f ../../tools/pre-commit.sh pre-commit` before commiting my files. Please let me know how to fix this.

No worries. Just run that command now from your branch and commit the formatted changes.

File: tensorflow_addons/losses/contrastive.py [hunk 1]
```diff
@@ -36,10 +35,19 @@ def contrastive_loss(
...
-    ```
+    >>> a = tf.constant([[1, 2],
+    ...                 [3, 4],
+    ...                 [5, 6]], dtype=tf.float16)
+    >>> b = tf.constant([[5, 9],
+    ...                 [3, 6],
+    ...                 [1, 8]], dtype=tf.float16)
+    >>> y_pred = tf.linalg.norm(a - b, axis=1)
+    >>> y_pred
+    <tf.Tensor: shape=(3,), dtype=float16, numpy=
```
Comment (bhack) [inline tensorflow_addons/losses/contrastive.py:46]: Can you try to use `# doctest: +NORMALIZE_WHITESPACE`

File: tensorflow_addons/losses/focal_loss.py [hunk 2]
```diff
@@ -37,22 +36,18 @@ class SigmoidFocalCrossEntropy(LossFunctionWrapper):
...
-    print('Loss: ', loss.numpy())  # Loss: [6.8532745e-06,
-                                            1.9097870e-04,
-                                            2.0559824e-05]
-    ```
+    >>> fl = tfa.losses.SigmoidFocalCrossEntropy()
+    >>> loss = fl(
+    ...    y_true = [[1.0], [1.0], [0.0]],
+    ...    y_pred = [[0.97], [0.91], [0.03]])
+    >>> loss
+    <tf.Tensor: shape=(3,), dtype=float32, numpy=
```
Comment (bhack) [inline tensorflow_addons/losses/focal_loss.py:44]: Can you try to use `# doctest: +NORMALIZE_WHITESPACE`

File: tensorflow_addons/losses/npairs.py [hunk 3]
```diff
@@ -33,10 +33,21 @@ def npairs_loss(y_true: TensorLike, y_pred: TensorLike) -> tf.Tensor:
...
-    ```
+    >>> a = tf.constant([[1, 2],
+    ...                 [3, 4],
+    ...                 [5, 6]], dtype=tf.float16)
+    >>> b = tf.constant([[5, 9],
+    ...                 [3, 6],
+    ...                 [1, 8]], dtype=tf.float16)
+    >>> y_pred = tf.matmul(a, b, transpose_a=False, transpose_b=True)
+    >>> y_pred
+    <tf.Tensor: shape=(3, 3), dtype=float16, numpy=
```
Comment (bhack) [inline tensorflow_addons/losses/npairs.py:44]: Can you try to use `# doctest: +NORMALIZE_WHITESPACE`

File: tensorflow_addons/losses/npairs.py [hunk 4]
```diff
@@ -89,10 +100,21 @@ def npairs_multilabel_loss(y_true: TensorLike, y_pred: TensorLike) -> tf.Tensor:
...
```
Comment (bhack) [inline tensorflow_addons/losses/npairs.py:111]: Can you try to use `# doctest: +NORMALIZE_WHITESPACE`

File: tensorflow_addons/losses/npairs.py [hunk 5]
```diff
@@ -139,10 +161,21 @@ class NpairsLoss(tf.keras.losses.Loss):
...
```
Comment (bhack) [inline tensorflow_addons/losses/npairs.py:172]: Can you try to use `# doctest: +NORMALIZE_WHITESPACE`

File: tensorflow_addons/losses/npairs.py [hunk 6]
```diff
@@ -184,10 +217,21 @@ class NpairsMultilabelLoss(tf.keras.losses.Loss):
...
```
Comment (bhack) [inline tensorflow_addons/losses/npairs.py:228]: Can you try to use `# doctest: +NORMALIZE_WHITESPACE`

File: tensorflow_addons/losses/contrastive.py (hunk 1 above)
Comment (Harsh188) [reply tensorflow_addons/losses/contrastive.py:46]: @bhack I just tried adding `# doctest: +NORMALIZE_WHITESPACE` but it seems to make no difference. (Maybe I'm using it wrong?) The interesting thing is that these tests pass for other files (`kappa_loss.py` &  `npairs.py`) where I utilize the same format.

Comment (bhack) [reply tensorflow_addons/losses/contrastive.py:46]: `kappa_loss.py` is single line.

Comment (bhack) [reply tensorflow_addons/losses/contrastive.py:46]: You find some example in https://pymotw.com/2/doctest/

**Review by bhack — DISMISSED** [review]

File: tensorflow_addons/losses/kappa_loss.py [hunk 7]
```diff
@@ -111,8 +111,8 @@ def __init__(
             self.weight_mat = (col_mat - row_mat) ** 2
 
     def call(self, y_true, y_pred):
-        y_true = tf.cast(y_true, dtype=self.col_label_vec.dtype)
-        y_pred = tf.cast(y_pred, dtype=self.weight_mat.dtype)
+        y_true = tf.cast(y_true, self.col_label_vec.dtype)
+        y_pred = tf.cast(y_pred, self.weight_mat.dtype)
```
Comment (WindQAQ) [inline tensorflow_addons/losses/kappa_loss.py:115]: Can you revert the changes here?

File: tensorflow_addons/losses/kappa_loss.py [hunk 8]
```diff
@@ -126,7 +126,7 @@ def call(self, y_true, y_pred):
         pred_dist = tf.reduce_sum(y_pred, axis=0, keepdims=True)
         w_pred_dist = tf.matmul(self.weight_mat, pred_dist, transpose_b=True)
         denominator = tf.reduce_sum(tf.matmul(label_dist, w_pred_dist))
-        denominator /= tf.cast(batch_size, dtype=denominator.dtype)
+        denominator /= tf.cast(batch_size, denominator.dtype)
```
Comment (WindQAQ) [inline tensorflow_addons/losses/kappa_loss.py:129]: ditto

File: tensorflow_addons/losses/contrastive.py [hunk 9]
```diff
@@ -15,10 +15,9 @@
 """Implements contrastive loss."""
 
 import tensorflow as tf
-
+from typeguard import typechecked
 from tensorflow_addons.utils.keras_utils import LossFunctionWrapper
 from tensorflow_addons.utils.types import TensorLike, Number
-from typeguard import typechecked
```
Comment (WindQAQ) [inline tensorflow_addons/losses/contrastive.py:21]: I would say
```python
import tensorflow as tf
from typeguard import typechecked
# empty line here
from tensorflow_addons.utils.keras_utils import LossFunctionWrapper
from tensorflow_addons.utils.types import TensorLike, Number
```

File: tensorflow_addons/losses/focal_loss.py [hunk 10]
```diff
@@ -16,10 +16,9 @@
 
 import tensorflow as tf
 import tensorflow.keras.backend as K
-
+from typeguard import typechecked
 from tensorflow_addons.utils.keras_utils import LossFunctionWrapper
 from tensorflow_addons.utils.types import FloatTensorLike, TensorLike
-from typeguard import typechecked
```
Comment (WindQAQ) [inline tensorflow_addons/losses/focal_loss.py:22]:
```python
import tensorflow as tf
import tensorflow.keras.backend as K
from typeguard import typechecked

from tensorflow_addons.utils.keras_utils import LossFunctionWrapper
from tensorflow_addons.utils.types import FloatTensorLike, TensorLike
```

File: tensorflow_addons/losses/giou_loss.py [hunk 11]
```diff
@@ -14,11 +14,11 @@
 # ==============================================================================
 """Implements GIoU loss."""
 
+from typing import Optional
 import tensorflow as tf
+from typeguard import typechecked
 from tensorflow_addons.utils.keras_utils import LossFunctionWrapper
 from tensorflow_addons.utils.types import TensorLike
```
Comment (WindQAQ) [inline tensorflow_addons/losses/giou_loss.py:21]:
```python
from typing import Optional

import tensorflow as tf
from typeguard import typechecked

from tensorflow_addons.utils.keras_utils import LossFunctionWrapper
from tensorflow_addons.utils.types import TensorLike
```

File: tensorflow_addons/losses/npairs.py [hunk 12]
```diff
@@ -16,8 +16,8 @@
 
 import tensorflow as tf
 
-from tensorflow_addons.utils.types import TensorLike
 from typeguard import typechecked
+from tensorflow_addons.utils.types import TensorLike
```
Comment (WindQAQ) [inline tensorflow_addons/losses/npairs.py:20]:
```python
import tensorflow as tf
from typeguard import typechecked

from tensorflow_addons.utils.types import TensorLike
```

**Review by WindQAQ — CHANGES_REQUESTED** [review]

**Review by WindQAQ — APPROVED** [review]
